code/game/turfs/turf_changing.dm

```diff
@@ -46,7 +53,36 @@
 	// So we call destroy.
 	qdel(src)
 
-	var/turf/simulated/W = new N( locate(src.x, src.y, src.z) )
+	var/turf/simulated/W = new N(src)
+
+#ifndef AO_USE_LIGHTING_OPACITY
+	// If we're using opacity-based AO, this is done in recalc_atom_opacity().
+	if (permit_ao)
+		regenerate_ao()
+#endif
+
+	recalc_atom_opacity()
+	lighting_overlay = old_lighting_overlay
+	if (lighting_overlay && lighting_overlay.loc != src)
+		// This is a hack, but I can't figure out why the fuck they're not on the correct turf in the first place.
+		lighting_overlay.forceMove(src, harderforce = TRUE)
+		
+	affecting_lights = old_affecting_lights
+	corners = old_corners
+
+	if ((old_opacity != opacity) || (dynamic_lighting != old_dynamic_lighting) || force_lighting_update)
+		reconsider_lights()
+
+	if (dynamic_lighting != old_dynamic_lighting)
+		if (dynamic_lighting)
+			lighting_build_overlay()
+		else
+			lighting_clear_overlay()
+
+	for (var/turf/space/S in RANGE_TURFS(1, src))
```

Comment:
Isn't starlight a config option/otherwise identifiable? Might we worth skipping this if possible.